mdast-util-to-hast types conflict with hast-util-raw

### Initial checklist

- [X] I read the [support docs](https://github.com/syntax-tree/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/syntax-tree/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/syntax-tree/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Asyntax-tree&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

mdast-util-to-hast@12.2.1 and hast-util-raw@7.2.2

### Link to runnable example

https://github.com/Methuselah96/test-mdast-raw

### Steps to reproduce

1. Add `react-markdown` and `rehype-raw` as dependencies in a project with `skipLibCheck` set to `false`.
2. Add source code that imports the packages.
3. Run type-checking.

### Expected behavior

No type errors.

### Actual behavior

Type errors:

```
node_modules/hast-util-raw/complex-types.d.ts:9:5 - error TS2717: Subsequent property declarations must have the same type.  Property 'raw' must be of type 'Raw', but here has type 'Raw'.

9     raw: Raw
      ~~~

  node_modules/mdast-util-to-hast/complex-types.d.ts:10:5
    10     raw: Raw
           ~~~
    'raw' was also declared here.

node_modules/hast-util-raw/complex-types.d.ts:13:5 - error TS2717: Subsequent property declarations must have the same type.  Property 'raw' must be of type 'Raw', but here has type 'Raw'.

13     raw: Raw
       ~~~

  node_modules/mdast-util-to-hast/complex-types.d.ts:14:5
    14     raw: Raw
           ~~~
    'raw' was also declared here.


Found 2 errors in the same file, starting at: node_modules/hast-util-raw/complex-types.d.ts:9
```

### Affected runtime and version

N/A

### Affected package manager and version

All

### Affected OS and version

N/A

### Build and bundle tools

_No response_

--- a/complex-types.d.ts
+++ b/complex-types.d.ts
@@ -1,10 +1,9 @@
 import type {Literal} from 'hast'
 
-export type Raw = {
+export interface Raw extends Literal {
   type: 'raw'
-} & Literal
+}
 
-/* eslint-disable @typescript-eslint/consistent-type-definitions */
 declare module 'hast' {
   interface RootContentMap {
     raw: Raw
@@ -14,4 +13,3 @@ declare module 'hast' {
     raw: Raw
   }
 }
-/* eslint-enable @typescript-eslint/consistent-type-definitions */

--- a/complex-types.d.ts
+++ b/complex-types.d.ts
@@ -1,5 +1,6 @@
 import type {Literal} from 'hast'
 
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
 export interface Raw extends Literal {
   type: 'raw'
 }
@@ -13,3 +14,4 @@ declare module 'hast' {
     raw: Raw
   }
 }
+/* eslint-enable @typescript-eslint/consistent-type-definitions */

--- a/complex-types.d.ts
+++ b/complex-types.d.ts
@@ -1,17 +1,18 @@
 import type {Literal} from 'hast'
 
-/* eslint-disable @typescript-eslint/consistent-type-definitions */
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
 export interface Raw extends Literal {
   type: 'raw'
 }
 
 declare module 'hast' {
+  // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
   interface RootContentMap {
     raw: Raw
   }
 
+  // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
   interface ElementContentMap {
     raw: Raw
   }
 }
-/* eslint-enable @typescript-eslint/consistent-type-definitions */

